ms_active_directory/core/ad_session.py

```diff
@@ -263,6 +263,92 @@ def _create_object(self, object_dn: str, object_classes: List[str], account_attr
+        :returns: an ADUser object representing the user.
+        :raises: ObjectCreationException if we fail to create the user for a reason unrelated to what we can
+                 easily validate in advance (e.g. permission issue)
+        """
+        logger.debug('Request to create user in domain %s with the following attributes: username=%s, '
+                     'first_name=%s, last_name=%s, object_location=%s, number of additional attributes specified: %s',
+                     self.domain_dns_name, first_name, last_name, object_location, len(additional_user_attributes))
+
+        if self.object_exists_in_domain_with_attribute(ldap_constants.AD_ATTRIBUTE_SAMACCOUNT_NAME, username):
+            raise ObjectCreationException('An object already exists with sAMAccountName {} so a user may not be '
+                                          'created with the username {}'.format(username, username))
+
+        # get or normalize our object location. the end format is as a relative distinguished name
+        # TODO: create helper function that normalizes an object location
+        if object_location is None:
+            object_location = ldap_constants.DEFAULT_USER_GROUP_LOCATION
+        else:
+            object_location = ldap_utils.normalize_object_location_in_domain(object_location, self.domain_dns_name)
+            # make sure our location exists
+            if ldap_utils.is_dn(object_location):
+                location_obj = self.find_object_by_distinguished_name(object_location)
+            else:
+                location_obj = self.find_object_by_canonical_name(object_location)
+            if location_obj is None:
+                raise ObjectCreationException('The user location {} cannot be found in the domain.'
+                                              .format(object_location))
+            # make sure our location is a container
+            is_container_or_ou = (location_obj.is_of_object_class(ldap_constants.ORGANIZATIONAL_UNIT_OBJECT_CLASS)
+                                  or location_obj.is_of_object_class(ldap_constants.CONTAINER_OBJECT_CLASS))
+            if not is_container_or_ou:
+                raise ObjectCreationException('The specified user location {} exists, but is not a container or an '
+                                              'organizational unit, and so a group cannot be created there.'
+                                              .format(location_obj.distinguished_name))
+            # make sure that going forward we have an LDAP-style relative distinguished name for our
+            # location
+            object_location = ldap_utils.normalize_object_location_in_domain(location_obj.distinguished_name,
+                                                                             self.domain_dns_name)
+        # construct display name from first and last names
+        display_name = first_name + ' ' + last_name
```

Comment:
since this becomes the common name as well, it'd be good to call out in the docstring what the default common name will be if unspecified (i know some groups do `CN=lastName, firstName` as their common names, so it'd be good to clarify that default)

Reply:
Should we add a default parameter for the common name, that if unspecified defaults to `first_name + ' ' + last_name`? I'll make sure to put in the docstring the default behavior.

Reply:
sounds good to me!

Reply:
Done.